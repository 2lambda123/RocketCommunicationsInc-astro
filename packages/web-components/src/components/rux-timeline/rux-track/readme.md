# rux-track

<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description | Type                  | Default |
| -------- | --------- | ----------- | --------------------- | ------- |
| `track`  | `track`   | the id      | `number \| undefined` | `0`     |


## Shadow Parts

| Part     | Description |
| -------- | ----------- |
| `"cell"` |             |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
