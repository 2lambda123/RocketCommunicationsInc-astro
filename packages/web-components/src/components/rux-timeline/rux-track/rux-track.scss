:host {
    display: contents;
}

.rux-track {
    display: contents;
    background: var(--color-surface);
}
.rux-track__header {
    padding: 11px 8px 13px 8px;
    background: var(--color-surface);
    border-bottom: 2px solid var(--color-background);
    display: flex;
    align-items: center;
    font-family: var(--font-body-1-font-family);
    font-size: var(--font-body-1-font-size);
    font-weight: var(--font-body-1-font-weight);
    letter-spacing: var(--font-body-1-letter-spacing);
}
.rux-track__header::slotted(*) {
    grid-column: header / time-0000;
}

::slotted(rux-time-region) {
    background: blue;
    // border: 2px solid red;
    // grid-row: 1;
}

.cell {
    border: 1px solid red;
    border-right: 0;
    // grid-column: span 60;
    z-index: 9999;
    grid-column: span 1;
}
