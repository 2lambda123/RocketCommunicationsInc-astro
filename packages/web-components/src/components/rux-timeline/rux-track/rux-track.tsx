import { Element, State, Component, Prop, Host, h } from '@stencil/core'

let id = 1
@Component({
    tag: 'rux-track',
    styleUrl: 'rux-track.scss',
    shadow: true,
    // styles: `
    //   :host, rux-track {
    //     background: red;
    //   }
    //   ::slotted(rux-time-region) {
    //     grid-row: ${id};
    //   }
    // `
})
export class RuxTrack {
    @Element() el!: HTMLRuxTrackElement
    @State() timelineColumns = 0
    private slotContainer?: HTMLElement

    // private _trackId = `rux-track-${++id}`

    /**the id */
    @Prop({ reflect: true }) track?: number = 0

    connectedCallback() {
        this._handleSlotChange = this._handleSlotChange.bind(this)
    }

    componentWillLoad() {
        // const timeline = this.el.closest('rux-timeline')
        this.initializeRows()

        const timeline = this.el.closest('rux-timeline')
        console.log('timeline', timeline)

        timeline?.fetchColumns().then((r) => {
            this.timelineColumns = r
        })
    }

    initializeRows() {
        const childNodes = this.el.childNodes
        const children = Array.prototype.filter.call(
            childNodes,
            (node) => node.nodeType == Node.ELEMENT_NODE
        )

        children.forEach((el) => {
            el.track = this.track
        })
    }

    private _handleSlotChange() {
        const slot = this.slotContainer?.querySelector(
            'slot'
        ) as HTMLSlotElement

        const assignedElements = slot.assignedElements({
            flatten: true,
        }) as HTMLElement[]

        //@ts-ignore
        assignedElements.map((el, index) => {
            //@ts-ignore
            // el.track = 5
            // el.trackId = ++index
            // el.setAttribute('track-id', 10)
        })
    }

    getCol(index: number) {
        if (index === 0) {
            return 2
        } else {
            return index + 2
        }
    }

    renderDebug() {
        return (
            <div style={{ display: 'contents' }}>
                {[...Array(this.timelineColumns)].map((x: any, i: any) => (
                    <div
                        style={{
                            gridRow: `${this.track}`,
                            gridColumn: `${i + 2} / ${++i + 2}`,
                        }}
                        class={{
                            cell: true,
                            marker: i % 60 === 0,
                        }}
                        part="cell"
                    ></div>
                ))}
            </div>
        )
    }

    render() {
        return (
            <Host>
                <div class="rux-track" ref={(el) => (this.slotContainer = el)}>
                    <div
                        class="rux-track__header"
                        style={{
                            gridRow: `${this.track}`,
                        }}
                    >
                        <slot name="label"></slot>
                    </div>

                    <slot onSlotchange={this._handleSlotChange}></slot>
                </div>
                {this.renderDebug()}
            </Host>
        )
    }
}
