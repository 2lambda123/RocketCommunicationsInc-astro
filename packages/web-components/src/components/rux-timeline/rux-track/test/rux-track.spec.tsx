import { newSpecPage } from '@stencil/core/testing'
import { RuxTrack } from '../rux-track'

describe('rux-track', () => {
    it('renders', async () => {
        const page = await newSpecPage({
            components: [RuxTrack],
            html: `<rux-track></rux-track>`,
        })

        expect(page.root).toMatchSnapshot()
    })
})
