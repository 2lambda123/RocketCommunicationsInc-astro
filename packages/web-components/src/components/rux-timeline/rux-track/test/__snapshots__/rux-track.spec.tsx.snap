// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rux-track renders 1`] = `
<rux-track track="0">
  <mock:shadow-root>
    <div class="rux-track">
      <div class="rux-track__header" style="grid-row: 0;">
        <slot name="label"></slot>
      </div>
      <slot></slot>
    </div>
  </mock:shadow-root>
</rux-track>
`;
